### PR TITLE
QCQMC Part 2

### PR DESCRIPTION
**Pull request by fdmalone**
Adds the quaff dependency. Quite a big commit.

### CODE REVIEW

**Comment by fdmalone** [comment]
Depends on #342

**Comment by fdmalone** [comment]
@wjhuggins, PTAL at the very minimal README / suggest something more to say. It's not really clear from a cursory scan of the code if there's a useful minimal example to show some useful features of the library.

**Comment by fdmalone** [comment]
@mpharrigan PTAL

File: recirq/third_party/__init__.py
Comment (mpharrigan) [inline recirq/third_party/__init__.py:1]: I think `third_party/` is supposed to go at the repository root and not be a python module itself, i.e. no `__init__.py`. and then futz around with the requirements or PYTHONPATH to pick up the `quaff` package

File: third_party/quaff/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# QUAFF
+
+Quaff is a Python library for manipulating, optimizing, and compiling Clifford gates.
```
Comment (fdmalone) [inline third_party/quaff/README.md:3]: @wjhuggins PTAL at this and make sure it's accurate please

**Comment by fdmalone** [comment]
@mpharrigan PTAL I migrated it to a top level third_party folder and added a note in the README about the separate installation.

**Comment by fdmalone** [comment]
cc @dstrain115 we decided adding the dependency to third_party was probably the best solution

File: third_party/quaff/README.md (hunk 1 above)
Comment (wjhuggins) [reply third_party/quaff/README.md:3]: LGTM

**Comment by fdmalone** [comment]
I cannot merge this myself

File: .github/workflows/pythonpackage.yml [hunk 2]
```diff
@@ -50,11 +51,17 @@ jobs:
           # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
           flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
 
+      - name: Test third_party 
+        run: |
+          pip install pytest
+          pytest third_party/quaff
```
Comment (mpharrigan) [inline .github/workflows/pythonpackage.yml:57]: you can probably remove this in a following PR when we have tests that exercise the quaff code path. We don't usually run tests for external packages

**Review by mpharrigan — APPROVED** [review]

**Pull request merged**